bread/layout/components/datatable.py

```diff
@@ -77,79 +34,63 @@ def from_modelfield(
             else None,
         )
 
+    def as_header_cell(self, orderingurlparameter="ordering"):
+        headcontent = hg.SPAN(self.header, _class="bx--table-header-label")
+        if self.sortingname:
+            headcontent = hg.BUTTON(
+                headcontent,
+                Icon("arrow--down", _class="bx--table-sort__icon", size=16),
+                Icon(
+                    "arrows--vertical",
+                    _class="bx--table-sort__icon-unsorted",
+                    size=16,
+                ),
+                _class=hg.BaseElement(
+                    "bx--table-sort ",
+                    sortingclass_for_column(orderingurlparameter, self.sortingname),
+                ),
+                data_event="sort",
+                title=self.header,
+                **sortinglink_for_column(orderingurlparameter, self.sortingname),
+            )
+        return hg.TH(headcontent, **getattr(self.header, "td_attributes", {}))
+
 
-class DataTable(hg.BaseElement):
+class DataTable(hg.TABLE):
     SPACINGS = ["default", "compact", "short", "tall"]
 
     def __init__(
         self,
         columns: List[DataTableColumn],
-        row_iterator,
-        rowvariable="row",
-        spacing="default",
-        orderingurlparameter="ordering",
-        zebra=False,
+        row_iterator: Union[hg.Lazy, Iterable, hg.Iterator],
+        orderingurlparameter: str = "ordering",
+        rowvariable: str = "row",
+        spacing: str = "default",
+        zebra: bool = False,
+        **kwargs: dict,
     ):
-        """columns: list of DataTableColumn(header, cell, sortingname, enable_row_click)
-        row_iterator: python iterator of htmlgenerator.Lazy object which returns an iterator
-        rowvariable: name of the current object passed to childrens context
-        if the header_expression/row_expression has an attribute td_attributes it will be used as attributes for the TH/TD elements (necessary because sometimes the content requires additional classes on the parent element)
-        sortingname: value for the URL parameter 'orderingurlparameter', None if sorting is not allowed
-        spacing: one of "default", "compact", "short", "tall"
-        zebra: alternate row colors
+        """A carbon DataTable element
+
+        :param columns: Column definitions
+        :param row_iterator: Iterator which yields row objects. If this is a hg.Iterator instance it will be used for the table body, otherwise a default Iterator will be used to render the column cells.
```

Comment:
I am not sure if it would help to mention also the `hg.Lazy` option here or if I just need to understand it better and then it will be clear ;)

Reply:
Now I read the original comment saying `...or htmlgenerator.Lazy object which returns an iterator`. Maybe we can just put that one back in and then it's clear.

Reply:
Okay, will add it again. Maybe we need a better way to mark support for ```Lazy``` values.

Lazy values are just values which are only evaluated very late, meaning at the time when the whole tree is serialized to an HTML string. A value which is lazy-generated cannot be modified or "queried" in the tree, that is why we want to keep them as rather small units. In most cases the lazy values mark that a value is only "extractable" or accessible while having an http-request from the framework available. In general these "values" are accessible through the context, which we passe when we call "render" on the html element tree, where we pass the django-context which contains at least the request object and often more (like the current database object).